### DISCUSSION TITLE
camera detecrion

### DISCUSSION THREAD

**Discussion by eriwm** [datazuul/eazy-bookscanner#6] [Q&A]
I have installed eazy bookscanner on a ubuntu 20.04 platform but it does not detect two chdk enabled cannon cameras connected to the usb ports. I have used these cameras on a chdkptp installation and they both work ok.  Any suggestions as to how I can go about fixing this. Thank You

**Comment by datazuul** [comment]
Thank you for testing. As eazy-bookscanner is based on https://github.com/acamilo/CHDK-PTP-Java each new Camera needs its own Class specifiying the "click"s needed for zooming etc., see: https://github.com/acamilo/CHDK-PTP-Java/tree/master/src/main/java/chdk/ptp/java/camera and https://github.com/acamilo/CHDK-PTP-Java/blob/master/src/main/java/chdk/ptp/java/SupportedCamera.java .

So checking out the library and eazy-bookscanner locally, adding specs/code for your camera to library, compiling it and recompiling eazy-bookscanner with the changed library should be enough. Is it possible for you to test it?

**Reply by datazuul** [reply]
see also: http://www.linux-usb.org/usb.ids for your camera's id:

323f  PowerShot A810

**Reply by eriwm** [reply]
Thank you kindly for replying so promptly. I have been trying for many weeks to get my cameras working with spreads but with very inconsistent results. I am not skilled in any programing languages but am always ready to try and learn something new so forgive me if some of my questions are pretty basic. I understand from your reply that I need to do the following-
1. Create a new library file called A810.java and add it to the library. This file must contain the necessary command for focus etc. for the A810.
2.  I have to edit the SupportedCamera.java file and add -
import chdkptp.java camera A810Camera; say around line 7
and 
add in public enum Supported Camera{A810 (0x04a9), 0x323f), A810Camera.class},
If my assumption is correct then my next hurdle is where in all the eazybookscanner folders can I find the library files and the SupportedCamera file.

For a first attempt I would try and use the contents of the A220 file just to see if I can get my camera recognized and then sort out the control functions afterwards.
Do I need any software tools to edit the files and do the recompiling.
Thank You

**Comment by datazuul** [comment]
Hi, yes the steps are right. Unfortunately it needs some build tool (java
jdk and maven)... I will be at home on saturday evening, then I could
implement a first version for you for testing...

ericwm ***@***.***> schrieb am Fr., 27. Mai 2022, 17:17:

**Reply by eriwm** [reply]
Opened SupportedCamera.java with jd-gui and edited it in text editor to include A810 Camera also opened A2200.java Camera the same way and changed all references of A2200 to A810 as per files below. Am i moving in the right direction? (changed file extensions from .java to .txt for upload)
[A810Camera.txt](https://github.com/datazuul/eazy-bookscanner/files/8800743/A810Camera.txt)
[SupportedCamera.txt](https://github.com/datazuul/eazy-bookscanner/files/8800746/SupportedCamera.txt)

**Reply by datazuul** [reply]
This is correct. Unfortunately I have problems compiling eazy-bookscanner (that uses the changed library) with new JDK as the netbeans-platform used with eazy-bookscanner no longer is compatible with. I will have to migrate code to be compilable again with new JDK...

This might take some time (weeks?) depending of my spare time resources. I will do it, but no quick solution is possible.

Workaround would be to install JDK 8 again... perhaps I will try it the next day this (fast) way. Hope you are still patient and eager to test then...

**Comment by datazuul** [comment]
I just finished a first version using configuration instead Java code for each camera. To give you a configuration example for your camera, I need the clicks needed to set your camera to "autofocus" starting from different current focus mode:

0=auto, 1=MF, 3=infinite, 4=macro, 5=supermacro

Example: if current mode is "0" (=auto): no clicks are needed

If current mode is "macro", the following clicks are needed (for my A2200) to change to autofocus:

`
[LEFT, sleep1000, RIGHT, sleep500, SET, sleep1000]
`
So just take your camera an do manually change from "macro" to "autofocus/normal" using the camera's buttons and report the navigation steps you needed... the same for all other modes (if exist): 0=auto, 1=MF, 3=infinite, 4=macro, 5=supermacro

maybe the manual <https://www.canon.de/support/consumer_products/products/cameras/digital_compact/powershot_a_series/powershot_a810.html?type=manuals&language=> helps, too.

After reporting I will give you a proper configuration to test

**Reply by datazuul** [reply]
I just added a draft configuration for your camera and made a new pre-release: https://github.com/datazuul/eazy-bookscanner/releases/tag/v1.0.0-SNAPSHOT-20220622 please download binary and test it. I just copied the clicks from A2200.

**Reply by eriwm** [reply]
I did respond to your post directly but I am adding here for completeness. 
I downloaded pre-releaseversion and installed according to "readme" but new version does not detect my cameras. I attach my version of of SupportedCamera file and A810Camera file that I use in previous version which could detect my cameras and gave me some unstable  control of shooting. Pre-release screen is verty neat thanks.
[A810Camera.txt](https://github.com/datazuul/eazy-bookscanner/files/8998274/A810Camera.txt)
[SupportedCamera.txt](https://github.com/datazuul/eazy-bookscanner/files/8998276/SupportedCamera.txt)
.
